executor/runtime/docker/docker.go

```diff
@@ -1828,119 +1836,31 @@ func setupNetworking(burst bool, c runtimeTypes.Container, cred ucred) (cleanupF
-		case <-doneCh:
-			// The command exited, no need to stand at our perch ready to terminate it.
-			return
-		}
-		err2 := setupCommand.Process.Signal(unix.SIGTERM)
-		if err2 != nil {
-			log.WithError(err2).Error("Unable to send SIGTERM to setup command")
-		}
-		timer := time.NewTimer(30 * time.Second)
-		defer timer.Stop()
-		select {
-		case <-timer.C:
-		case <-doneCh:
-			// The command successfully exited
-			return
-		}
-		// The timer fired, and it's time to send the kill signal
-		log.Warn("Sending kill signal to setup command")
-		err2 = setupCommand.Process.Kill()
-		if err2 != nil {
-			log.WithError(err2).Error("Unable to send SIGKILL to allocation command")
-		}
-	}()
-
-	killTimer := time.AfterFunc(2*time.Minute, func() {
-		killCh <- struct{}{}
-	})
-
-	waitForKill := func() {
-		defer close(doneCh)
-		errCh <- setupCommand.Wait()
-	}
 
 	allocation := *c.VPCAllocation()
 	if err := json.NewEncoder(stdin).Encode(allocation); err != nil {
-		go waitForKill()
-		killCh <- struct{}{}
 		return nil, err
 	}
+
 	if err := json.NewDecoder(stdout).Decode(&result); err != nil {
-		go waitForKill()
-		killCh <- struct{}{}
 		return nil, fmt.Errorf("Unable to read json from pipe during setup-container: %+v", err)
 	}
 
-	go waitForKill()
-
-	if !killTimer.Stop() {
-		err = errors.New("Kill timer fired. Race condition")
-		log.WithError(err).Error("Accidentally killed the setup command, leaving us in a 'unknown' state")
-		return nil, err
-	}
-
 	if !result.Success {
-		killCh <- struct{}{}
 		return nil, fmt.Errorf("Network setup error: %s", result.Error)
 	}
 
-	switch g := (*allocation.Generation); g {
-	case vpcTypes.V1:
-		return func() error {
-			ctx, cancel := context.WithTimeout(context.Background(), 45*time.Second)
-			defer cancel()
-
-			killCh <- struct{}{}
-			var err2 error
-			select {
-			case err2 = <-errCh:
-			case <-ctx.Done():
-				return errors.Wrap(ctx.Err(), "Error waiting for v1 setup command to exit")
-			}
-			if err2 != nil {
-				log.WithError(err2).Error("Received error on termination of setup command")
-				return errors.Wrap(err2, "Could not teardown container networking")
-			}
-			return nil
-		}, nil
-	case vpcTypes.V3:
-		// No one should have read off errCh before us.
-		err = <-errCh
-		if err != nil {
-			killCh <- struct{}{}
-			return nil, errors.Wrap(err, "Error experienced when running V3 setup command")
-		}
-		f2, err := os.Open(netnsPath)
-		if err != nil {
-			return nil, errors.Wrap(err, "Unable to open container network namespace file")
-		}
-		return func() error {
-			return teardownCommand(f2, allocation)
-		}, nil
-	default:
-		err = fmt.Errorf("Unknown generation: %s", g)
-		killCh <- struct{}{}
-		log.WithError(err).Error("Received allocation with unknown generation")
-		return nil, err
+	f2, err := os.Open(netnsPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to open container network namespace file")
 	}
+	return func() error {
```

Comment:
This is interesting. Is this supposed to be called if anything goes wrong in the above?
If yes, I don't see how that is possible with when we return normal errors above, and the only thing I see deferred is the contex cancel function.